2018/002-update-channels.md

```diff
@@ -0,0 +1,123 @@
+update channels. This allows the separation of possibly disruptive updates
+into their own channels, and lets the admin choose a conservative or progressive
+update mechanism. The admin can also see what is inside an update to be able
+to decide when to apply it.
+
+### Proposed change
+
+The admin will have the ability of choosing when to switch to the next channel.
+A link to the release notes of the update channel is displayed and can be
+evaluated by the admin beforehand.
+
+The UI of velum needs to be adapted to enable a switch to a new channel.
+
+## Detailed RFC
+
+We can separate the updates into different channels. There are 2 types of channels:
+
+* maintenance channel: normal maintenance updates, which are safe and non-intrusive
+* new version channel: version updates which can contain intrusive updates
+
+This RFC proposes:
+
+* a front-end change in the velum UI to enable the admin to switch to a new version
+channel
+* a back-end change in salt to perform the channel switch on all cluster nodes
+
+### Proposed change (Detailed)
+
+#### Tools/Services
+
+We need a set of tools/services to indicate, whenever a maintenance update or a new version
+update is ready to be installed. They should run on each worker node to be more independent
+and decoupled from velum. If velum would be the initiator of the update check, it might lead
+to inconsistent and long execution runs in larger clusters.
+
+* "check for updates" service on each worker
+  - reports availability of maintenance updates to velum (via grains?)
+  - once per day e.g.
+  - decoupled from velum, runs independently, just reports
+* "new version" checker service on each worker
+  - check for new versions (migration targets) (via SCC? or own solution)
+    - report back to velum (via grains?)
+  - get the link to the release notes
+  - once per week or longer timeframe
+  - decoupled from velum, runs independently, just reports
+
+#### User Interface
+
+The user interface of velum needs to be adapted minimally.
+
+* [indirect] display of update content through [a link to the] release notes
+* display of update history via velum
+* button to switch to the next channel in velum
+* separate UI for updates and upgrades (new migration target)
+
+#### API
+
+Furthermore there needs to be an API available through velum to enable easy scripting
+
+* endpoints:
+  - GET:  list current channel maintenance updates (non disruptive)
+  - POST: trigger current channel maintenance updates
+  - GET:  list next channel (new version) updates
+      - A: disruptive update in the same caasp product/version (e.g. k8s minor version bump, container runtime)
+      - B: migration target once overlap support is EOL (next caasp product/version)
+  - POST: switch to the next channel
+  - GET:  list update history
+  - POST: trigger reboots
+
+#### General considerations
+
+It is essential to enforce to run all maintenance updates first, and then allow the channel
```

Comment:
I don't really like to enforce the customer to apply updates he don't need, because the following migration will immeadilty overwrite them. We should come up with a solution, where we can check, if the installed packages are good enough for the migration or not.